grafana/grafana
# Explore: Add caching for queries that are run from logs navigation

**What this PR does / why we need it**:
After unsuccessful caching 1.0 https://github.com/grafana/grafana/pull/33744, I am bringing for review **caching 2.0**. Better and faster! Instead of implementing caching on component level which caused issue with inspector, time range picker and results being out of sync, we are  implementing caching in redux. We are caching the last 5 results run from navigation/that open navigation and then when we are supposed to run the query, we always check the cache. If results are already cached, we use those instead of running query. 

**How to test:**
1. Open network tab
2. Run logs query (can be Loki, Elastic or test-data logs)
3. For new queries, you can see request in network tab, for cached you don't see it.
4. Check if time range picker and panel inspector and url have correct data.
